### PR TITLE
Add bcftools/pluginsetgt

### LINKED ISSUE #6717: new module: BCFTOOLS/PLUGINSETGT
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Further Information

_No response_

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by eweizy**
## PR checklist

Closes #6717

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/bcftools/pluginsetgt/main.nf [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    """
+    bcftools plugin setGT \\
+        --output ${prefix}.${extension} \\
+        ${regions_file} \\
+        ${targets_file} \\
+        $args \\
+        --threads $task.cpus \\
+        ${vcf} \\
+        -- \\
+        $args2
```
Comment (nvnieuwk) [inline modules/nf-core/bcftools/pluginsetgt/main.nf:44]: What's this doing?

**Review by nvnieuwk — COMMENTED** [review]
One small comment it LGTM otherwise

File: modules/nf-core/bcftools/pluginsetgt/main.nf (hunk 1 above)
Comment (eweizy) [reply modules/nf-core/bcftools/pluginsetgt/main.nf:44]: BCFtools plugins are run as bcftools [plugin NAME|+NAME] [OPTIONS] FILE  --  [PLUGIN OPTIONS]. Most plugins accept two types of parameters: general options shared by all plugins followed by a separator, and a list of plugin-specific options. I've set `args` to be the general options and `args2` to be the plugin-specific options.

Comment (nvnieuwk) [reply modules/nf-core/bcftools/pluginsetgt/main.nf:44]: Allright thanks! That seems reasonable then :)

Comment (nvnieuwk) [reply modules/nf-core/bcftools/pluginsetgt/main.nf:44]: Does it also work if you don't give any args in `args2`?

Comment (eweizy) [reply modules/nf-core/bcftools/pluginsetgt/main.nf:44]: It does not, at minimum the `-n, --new-gt TYPE` and ` -t, --target-gt TYPE ` plugin options need to be set for this plugin to work

Comment (eweizy) [reply modules/nf-core/bcftools/pluginsetgt/main.nf:48]: non-file mandatory command arguments added here https://github.com/nf-core/modules/pull/6718/commits/a5e6c0cdd17b2581b7a1b969926f2d8c0d08f3a9

**Review by nvnieuwk — APPROVED** [review]
Thank you! LGTM

**Pull request merged**